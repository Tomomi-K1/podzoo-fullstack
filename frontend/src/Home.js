import React, { useState, useEffect } from "react";
import axios from "axios";


function Home() {
    
  const [podcasts, setPodcasts] = useState(null);

  useEffect(function getPodcasts() {
    console.debug("get podcasts in useEffect");
    api();
  }, []);

    async function api(){
        console.log(`api call`)
        let result =await axios.get("http://localhost:3001/");
        setPodcasts(result.data.items);
    }
    
    console.log(podcasts)
    if(!podcasts){
        return <h1>loading...</h1>
    }
  
    return (
        <div className="Homepage">
        <h1> Pearch! </h1>    
         <h2>Episode</h2>
{podcasts.map(feed =>{
    return (
        <div className={feed.id}>
            <img src={feed.image} alt="podcast artwork" height={100} width={100}/>
            <h3>{feed.title}</h3>
            <p>{feed.description}</p>
            <figure>
            <figcaption>Listen to episode {feed.episode}</figcaption>
                <audio
                    controls
                    src={feed.enclosureUrl}>

                </audio>
            </figure>
        </div>
    ) 
 })}
    
        </div>
    );
  }
  
  export default Home;

  
// <h2>Episode</h2>
// {podcasts.map(feed =>{
//     return (
//         <div className={feed.id}>
//             <img src={feed.image} alt="podcast artwork" height={100} width={100}/>
//             <h3>{feed.title}</h3>
//             <p>{feed.description}</p>
//             <figure>
//             <figcaption>Listen to episode {feed.episode}</figcaption>
//                 <audio
//                     controls
//                     src={feed.enclosureUrl}>

//                 </audio>
//             </figure>
//         </div>
//     ) 
//  })}
  