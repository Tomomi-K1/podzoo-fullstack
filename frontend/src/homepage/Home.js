import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";

import removeTags from "../common/helper";
import Loader from "../common/Loader";

import PodApi from "../api/PodApi";

import PodcastList from "../podcasts/PodcastList";
import CategoryList from "../category/CategoryList";
import categories from "../category/categories";
// Material UI
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import SearchIcon from '@mui/icons-material/Search';

function Home() {
    
    const [podcasts, setPodcasts] = useState(null);
    const [searchTerm, setSearchTerm] = useState("");
    const navigate = useNavigate();

    useEffect(function getPodcasts() {
        console.debug("get podcasts in useEffect");
        api();
    }, []);

    async function api(){
        console.log(`api call at Home`)
        try{
            let result =await PodApi.getTrendingPodcasts();
            setPodcasts(result);
        }catch(err){
            console.log(`error when making api call`)
        }
    }

    function handleSubmit(evt){
        evt.preventDefault();
        navigate(`/search/${searchTerm}`);
    }

    function handleChange(evt){
        evt.preventDefault();
        let {value} =evt.target;
        setSearchTerm(value);
        console.log(searchTerm);
    }
    
    if(!podcasts){
        return <Loader />
    }
  
    return (
        <div className="Homepage">
            <form onSubmit={handleSubmit}>
            <Box sx={{display:'flex', width: '100%', justifyContent:'center', alignItems:'center', height:'100px', pt:'20px'}}>
            <TextField sx ={{width: '90%'}}
                placeholder='search'
                value={searchTerm}
                onChange={handleChange}
            >
                <SearchIcon />
            </TextField>
            </Box>
            </form>
            <Typography variant="h3" sx={{m:5, fontWeight:'bold'}}> Find by Categories</Typography>
                <CategoryList categories={categories}/>
        
            <Typography variant="h3" sx={{m:5, fontWeight:'bold'}}>Trending podcasts</Typography>
                <PodcastList podcasts={podcasts}/>
        </div>
        
    );
  }
  
  export default Home;

  
// <h2>Episode</h2>
// {podcasts.map(feed =>{
//     return (
//         <div className={feed.id}>
//             <img src={feed.image} alt="podcast artwork" height={100} width={100}/>
//             <h3>{feed.title}</h3>
//             <p>{feed.description}</p>
//             <figure>
//             <figcaption>Listen to episode {feed.episode}</figcaption>
//                 <audio
//                     controls
//                     src={feed.enclosureUrl}>

//                 </audio>
//             </figure>
//         </div>
//     ) 
//  })}
  