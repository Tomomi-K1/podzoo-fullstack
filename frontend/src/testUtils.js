import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };