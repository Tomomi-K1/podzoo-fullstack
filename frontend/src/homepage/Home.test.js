import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "./Home";
import { UserProvider } from "../testUtils";

it('renders', () => {
    render(
    <MemoryRouter>
        <Home />
    </MemoryRouter>
    );
})

it('matches snapshot: user logged in', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Home />
            </UserProvider>
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

it('matches snapshot when no user', () => {
    const { asFragment } = render(
        <MemoryRouter>
                <Home />
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

