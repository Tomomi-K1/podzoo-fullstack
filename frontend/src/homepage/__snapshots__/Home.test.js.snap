// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when no user 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Loading...
    </h3>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot: user logged in 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Loading...
    </h3>
  </div>
</DocumentFragment>
`;
